src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -86,5 +100,45 @@ public override void OnStopActivity(Activity activity, object payload)
             span.PutHttpStatusCode(response.StatusCode, response.HttpContext.Features.Get<IHttpResponseFeature>().ReasonPhrase);
             span.End();
         }
+
+        private static string GetUri(HttpRequest request)
+        {
+            if (request == null)
+            {
+                return string.Empty;
+            }
+
+            var builder = new StringBuilder();
+
+            builder.Append(request.Scheme).Append("://");
+
+            if (request.Host.HasValue)
+            {
+                builder.Append(request.Host.Value);
+            }
+            else
+            {
+                // HTTP 1.0 request with NO host header would result in empty Host.
+                // Use placeholder to avoid incorrect URL like "http:///"
+                builder.Append(UnknownHostName);
+            }
+
+            if (request.PathBase.HasValue)
+            {
+                builder.Append(request.PathBase.Value);
+            }
+
+            if (request.Path.HasValue)
+            {
+                builder.Append(request.Path.Value);
+            }
+
+            if (request.QueryString.HasValue)
+            {
+                builder.Append(request.QueryString);
+            }
+
+            return builder.ToString();
```

Comment:
why not request.ResuestUrl.ToString() (or whatever we have?)

Reply:
there is no request Url available. As far as I know